src/m365/entra/commands/app/app-permission-list.ts

```diff
@@ -0,0 +1,248 @@
+    };
+
+    const res = await request.get<{ value: { id: string }[] }>(requestOptions);
+
+    if (res.value.length === 0) {
+      throw `No Azure AD application registration with ID ${options.appId} found`;
+    }
+
+    return res.value[0].id;
+  }
+
+  private async getAppRegPermissions(appObjectId: string, permissionType: string, logger: Logger): Promise<ApiPermission[]> {
+    const requestOptions: CliRequestOptions = {
+      url: `${this.resource}/v1.0/myorganization/applications/${appObjectId}`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    const application = await request.get<Application>(requestOptions);
+    const requiredResourceAccess = application.requiredResourceAccess as RequiredResourceAccess[];
+
+    if (requiredResourceAccess.length === 0) {
+      return [];
+    }
+
+    const servicePrincipalsToResolve: ServicePrincipalInfo[] =
+      requiredResourceAccess.map(resourceAccess => {
+        return {
+          appId: resourceAccess.resourceAppId as string
+        };
+      });
+    const servicePrincipals = await Promise
+      .all(servicePrincipalsToResolve.map(servicePrincipalInfo =>
+        this.getServicePrincipal(servicePrincipalInfo, permissionType, logger) as ServicePrincipal));
+
+    const apiPermissions: ApiPermission[] = [];
+    requiredResourceAccess.forEach(requiredResourceAccess => {
+      const servicePrincipal = servicePrincipals
+        .find(servicePrincipal => servicePrincipal?.appId === requiredResourceAccess.resourceAppId as string);
+      const resourceName = servicePrincipal?.displayName as string ?? requiredResourceAccess.resourceAppId as string;
+      (requiredResourceAccess.resourceAccess as ResourceAccess[]).forEach(permission => {
+        if (permissionType === 'application' && permission.type === 'Scope') { return; }
+        if (permissionType === 'delegated' && permission.type === 'Role') { return; }
+
+        apiPermissions.push({
+          resource: resourceName,
+          resourceId: requiredResourceAccess.resourceAppId,
+          permission: this.getPermissionName(permission.id as string, permission.type as string, servicePrincipal),
+          type: permission.type === 'Role' ? 'Application' : 'Delegated'
+        });
+      });
+    });
+
+    return apiPermissions;
+  }
+
+  private getPermissionName(permissionId: string, permissionType: string, servicePrincipal: ServicePrincipal | undefined): string {
+    if (!servicePrincipal) {
+      return permissionId;
+    }
+
+    if (permissionType === 'Role') {
+      return (servicePrincipal.appRoles as AppRole[])
+        .find(appRole => appRole.id === permissionId)?.value as string ?? permissionId;
+    }
+    else {
+      // permissionType === 'Scope'
+      return (servicePrincipal.oauth2PermissionScopes as PermissionScope[])
+        .find(permissionScope => permissionScope.id === permissionId)?.value as string ?? permissionId;
+    }
```

Comment:
actually if the `if` block have `return` statement it makes no sense to do the `else`.
In this case we may do something like 
```suggestion
    if (permissionType === 'Role') {
      return (servicePrincipal.appRoles as AppRole[])
        .find(appRole => appRole.id === permissionId)?.value as string ?? permissionId;
    }
    
      return (servicePrincipal.oauth2PermissionScopes as PermissionScope[])
        .find(permissionScope => permissionScope.id === permissionId)?.value as string ?? permissionId;
```

which logically is the same and is more readable as we reduce the amount of nesting code